src/arch/ucx/machine-onesided.C

```diff
@@ -70,7 +70,7 @@ void LrtsDeregisterMem(const void *ptr, void *info, int pe, unsigned short int m
 
     UCX_LOG(4, " %p, PE %d, info %p, memh %d", ptr, pe, ucxInfo, ucxInfo->memh);
```

Comment:
Currently, the Charm++ layer assumes that both source and destination memory is registered and invokes CmiDeregisterMem(which calls LrtsDeregisterMem). 

However, from our conversation yesterday, I found out that the UCX layer only requires peer remote memory to be registered. (and not local memory).

For that reason, /examples/charm++/zerocopy/entry_method_api/unreg/simpleZeroCopy is failing with a segfault inside this method. (Because it is trying to de-register a region that was not registered). 

Is there a way to find out with the UCX layer if the memory is registered? And only then de-register? (to bypass that de-registration?). 

If not, I can modify the LrtsDeregisterMem API to have the charm layer send a flag and not de-register in specific cases.

Reply:
The flag option sounds optimal. UCX does not provide such an API, so I'd need to hash all addresses in UCX machine layer to be able to make a decision.